lib/expand.js

```diff
@@ -695,6 +697,47 @@ class Expander {
     return constraints;
   }
 
+  consolidateSubsetConstraint(element, value, constraint, previousConstraints) {
+    constraint = constraint.clone();
+    // If the constraint is actually on the target's value, update the path to explicitly include the target's value
+    if (constraint.onValue) {
+      const targetVal = this.constraintTargetValue(value, constraint.path);
+      // If it's a choice, check if the constraint is the choice or a subtype of the choice
+      if (targetVal instanceof models.ChoiceValue) {
+        const optionIdentifiers = targetVal.aggregateOptions.map(opt => opt.identifier);
+        const areValidOptions = constraint.subsetList.every(elem => {
+          return optionIdentifiers.some(optId => optId.equals(elem));
+        });
+        if (!areValidOptions) {
+          const targetLabel = this.constraintTargetLabel(value, constraint.path);
+          logger.error('Cannot constrain type of %s to %s. ERROR_CODE:12012', targetLabel, constraint.subsetList);
+          return previousConstraints;
+        }
+      }
+    }
+
+    const target = this.constraintTarget(value, constraint.path);
+
+    let constraints = previousConstraints;
+    const filtered = (new models.ConstraintsFilter(previousConstraints)).withPath(constraint.path).subset.constraints;
+    for (const previous of filtered) {
+      if (constraint.onValue != previous.onValue) {
+        continue;
+      }
+      // Check if the each element of the new subset has some base type in the previous subset
+      // Does allow elements in previous subset to serve as base type for multiple elements in new subset
```

Comment:
I don't see a test for this, and based on what I see in lines 707 - 716, I kind of doubt that child types work.  Do they?